mule-deployer/src/main/java/org/mule/tools/verification/fabric/RuntimeFabricDeploymentVerification.java

```diff
@@ -55,28 +58,44 @@ private RuntimeFabricDeploymentVerificationStrategy() {
 
     public Predicate<Deployment> isDeployed() {
       return (deployment) -> {
-        String deploymentId = getDeploymentId(deployment);
-        if (deploymentId == null) {
-          return false;
-        }
-        DeploymentDetailedResponse response = client.getDeployment(deploymentId);
-        if (response != null) {
-          if (StringUtils.equals(response.status, FAILED_STATUS)) {
-            throw new IllegalStateException("Deployment failed");
-          } else if (StringUtils.equals(response.status, APPLIED_STATUS)
-              || StringUtils.equals(response.status, STARTED_STATUS)) {
-            return true;
+        try {
+          String deploymentId;
+
+          deploymentId = getDeploymentId(deployment);
+
+          if (deploymentId == null) {
+            return false;
+          }
+          DeploymentDetailedResponse response = client.getDeployment(deploymentId);
+          if (response != null) {
+            if (StringUtils.equals(response.status, FAILED_STATUS)) {
+              throw new IllegalStateException("Deployment failed");
+            } else if (StringUtils.equals(response.status, APPLIED_STATUS)
+                || StringUtils.equals(response.status, STARTED_STATUS)) {
+              return true;
+            }
           }
+        } catch (DeploymentException e) {
+          e.printStackTrace();
+          return false;
         }
         return false;
       };
     }
 
-    private String getDeploymentId(Deployment deployment) {
+    private String getDeploymentId(Deployment deployment) throws DeploymentException {
+      RuntimeFabricDeployment deploymentRTF = (RuntimeFabricDeployment) deployment;
+      String targetName = deploymentRTF.getTarget();
+      JsonArray targets = client.getTargets();
+      String targetID = null;
+      if (targets != null) {
+        targetID = RequestBuilder.getTargetId(targets, targetName);
+      }
       if (deploymentId == null) {
         deployments = client.getDeployments();
         for (DeploymentGenericResponse dep : deployments) {
-          if (StringUtils.equals(dep.name, deployment.getApplicationName())) {
+          if (StringUtils.equals(dep.name, deploymentRTF.getApplicationName()) && (targets == null ||
+              StringUtils.equals(targetID, dep.target.targetId))) {
             deploymentId = dep.id;
```

Comment:
Acá setea el deploymentId y cuando se retorna el método se vuelve a setear otra vez la misma variable (en la línea 64)? Por ahí me estoy perdiendo algo.
Se puede cambiar levemente el nombre de la variable local al método, para no confundirla con la de la clase.

Reply:
these are different vars, the var used in the line 64 is a different one (local of that method)